src/tests/ftest/util/ior_utils.py

```diff
@@ -20,166 +20,319 @@
+                                "key: {}".format(name, item, sub_item, key))
+                else:
+                    raise IorFailed(
+                        "Error obtaining the IOR aggregate total from the {}: "
+                        "value: {}, split: {}".format(name, item, sub_item))
+
+        # Account for any replicas
+        try:
+            # Extract the replica quantity from the object class string
+            replica_qty = int(re.findall(r"\d+", self.daos_oclass.value)[0])
+        except (TypeError, IndexError):
+            # If the daos object class is undefined (TypeError) or it does not
+            # contain any numbers (IndexError) then there is only one replica
+            replica_qty = 1
+        finally:
+            total *= replica_qty
+
+        return total
+
+    def get_launch_command(self, basepath, processes, hostfile, runpath=None):
+        """Get the process launch command used to run IOR.
+
+        Args:
+            basepath (str): DAOS base path
+            processes (int): number of host processes
+            hostfile (str): file defining host names and slots
+            runpath (str, optional): Optional path to the mpirun/oretrun
+                command. Defaults to None.
+
+        Raises:
+            IorFailed: if an error occured building the IOR command
+
+        Returns:
+            str: ior launch command
+
+        """
+        with open(os.path.join(basepath, ".build_vars.json")) as afile:
+            build_paths = json.load(afile)
+        attach_info_path = os.path.join(basepath, "install/tmp")
+
+        if self.api.value == "MPIIO":
+            env = {
+                "CRT_ATTACH_INFO_PATH": attach_info_path,
+                "DAOS_POOL": self.daos_pool.value,
+                "MPI_LIB": "",
+                "DAOS_SVCL": self.daos_svcl.value,
+                "DAOS_SINGLETON_CLI": 1,
+                "FI_PSM2_DISCONNECT": 1,
+            }
+            export_cmd = [
+                "export {}={}".format(key, val) for key, val in env.items()]
+            mpirun_cmd = [
+                os.path.join(
+                    runpath if runpath else build_paths["OMPI_PREFIX"],
+                    "bin/mpirun"),
+                "-np {}".format(processes),
+                "--hostfile {}".format(hostfile),
+            ]
+            command = " ".join(mpirun_cmd + [self.__str__()])
+            command = "; ".join(export_cmd + [command])
+
+        elif self.api.value == "DAOS":
+            orterun_cmd = [
+                os.path.join(
+                    runpath if runpath else build_paths["OMPI_PREFIX"],
+                    "bin/orterun"),
+                "-np {}".format(processes),
+                "--hostfile {}".format(hostfile),
+                "--map-by node",
+                "-x DAOS_SINGLETON_CLI=1",
+                "-x CRT_ATTACH_INFO_PATH={}".format(attach_info_path),
+            ]
+            command = " ".join(orterun_cmd + [self.__str__()])
```

Comment:
So to be clear, this use of orterun should be removed and mpich's mpirun used for both `-a {MPIIO,DAOS}`.